### ISSUE TITLE
Problem writing a large file to ZipArchive

### ISSUE THREAD

**Issue by Bo0m21** [dotnet/runtime#101238]
### Description

The problem occurs when writing a large file to an archive.

### Reproduction Steps

1. Run this code
2. Have error
3. Fine

```
using (FileStream zipToOpen = new FileStream("release.zip", FileMode.OpenOrCreate))
{
    using (ZipArchive archive = new ZipArchive(zipToOpen, ZipArchiveMode.Update))
    {
        ZipArchiveEntry readmeEntry = archive.CreateEntry("Readme.txt");
        using (StreamWriter writer = new StreamWriter(readmeEntry.Open()))
        {
            for (int i = 0; i < 250000000; i++)
            {
                writer.WriteLine("Information about this package.");
                writer.WriteLine("========================");
            }
        }
    }
}
```

### Expected behavior

The code will run without errors

### Actual behavior

The code worked with an error
System.IO.IOException: 'Stream was too long.'

**Comment by PaulusParssinen** [comment]
What error?

**Comment by PaulusParssinen** [comment]
> > What error?
> 
> Updated issue.

Thanks! It seems like you're hitting `StreamWriter`'s internal buffer limit by trying to write over 2GB in one go. Depending on your you may want to periodically issue `StreamWriter.Flush` or chunk the data accordingly. You can find similar solutions to this simply by googling the error.

**Comment by PaulusParssinen** [comment]
> ...
> 
> Before posting the error here, I researched other similar problems and, of course, used Google. Could you please provide working code?

This entirely depends on your actual use-case but my first reaction would be do something along the lines:

```CSharp
using (FileStream zipToOpen = new FileStream("release.zip", FileMode.OpenOrCreate))
{
    using (ZipArchive archive = new ZipArchive(zipToOpen, ZipArchiveMode.Update))
    {
        ZipArchiveEntry readmeEntry = archive.CreateEntry("Readme.txt");
        using (StreamWriter writer = new StreamWriter(readmeEntry.Open()))
        {
            for (int i = 0; i < 250000000; i++)
            {
                writer.WriteLine("Information about this package.");
                writer.WriteLine("========================");
                if (i % 100_000 /* or something */ == 0) writer.Flush();
            }
        }
    }
}
``

**Comment by PaulusParssinen** [comment]
> ...
> 
> Could you please provide working code?

So, this seems to essentially be blocked by https://github.com/dotnet/runtime/issues/1544.

__IF__ you can rebuild the existing archive, you can avoid the in-memory `MemoryStream` buffer caused by `readMeEntry.Open()` by overwriting existing archive in write-only manner. Otherwise, you're blocked by #1544  

```cs
using (FileStream zipToOpen = new FileStream("release.zip", FileMode.OpenOrCreate, FileAccess.Write))
{
    using (ZipArchive archive = new ZipArchive(zipToOpen, ZipArchiveMode.Create))
    {
        ZipArchiveEntry readmeEntry = archive.CreateEntry("Readme.txt");
        using (StreamWriter writer = new StreamWriter(readmeEntry.Open()))
        {
            for (int i = 0; i < 250000000; i++)
            {
                writer.WriteLine("Information about this package.");
                writer.WriteLine("========================");
            }
        }
    }
}
```

**Comment by ericstj** [comment]
Duplicate of #1544
